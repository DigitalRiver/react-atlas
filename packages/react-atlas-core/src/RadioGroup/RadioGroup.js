import React, { cloneElement } from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class RadioGroup extends React.PureComponent {
  constructor(props) {
    super(props);
    // Initial state
    this.state = {
      checkedRadio: null
    };
  }

  componentWillMount() {
    React.Children.map(this.props.children, (child, index) => {
      if (
        child.props.defaultChecked ||
        (!this.state.checkedRadio && index === 0)
      ) {
        this.setState({ checkedRadio: child.props.value });
      }
    });
  }

  groupSetChecked = value => {
    this.setState({ checkedRadio: value });
  };

  render() {
    const {
      className,
      children,
      name,
      inline,
      inlineChildren,
      title,
      style
    } = this.props;

    const radioGroupStyles = cx({
      radioGroup: !inline,
      inline: inline
    });

    const radioButtons = React.Children.map(children, child => {
      child = cloneElement(child, {
        inline: inlineChildren,
        name: name,
        groupSetChecked: this.groupSetChecked,
        checked: this.state.checkedRadio === child.props.value
      });
      return child;
    });

    return (
      <div style={style} className={cx(className)} styleName={radioGroupStyles}>
        {title && (
          <div styleName={cx({ header: !inline })}>
            <span styleName={"headerFont"}>{title}</span>
          </div>
        )}
        {radioButtons}
      </div>
    );
  }
}

RadioGroup.propTypes = {
  /**
   * Anything that can be in a radio group. Almost always radio components alone.
   * @examples '<RadioGroup><Radio/><Radio/></RadioGroup>'
   */
  children: PropTypes.node.isRequired,
  /** An Object, array, or string of CSS classes to apply to RadioGroup.*/
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   * Form name for the element, this will set all Radio children the same form name (so they can't be selected at the same time).
   * @examples '<RadioGroup name="test"></RadioGroup>'
   */
  name: PropTypes.string,
  /**
   * Define the element title.
   * @examples '<RadioGroup title="Test"></RadioGroup>'
   */
  title: PropTypes.string,
  /**
   * Defines if the radio group should display as an inline element.
   * @examples '<RadioGroup inline></RadioGroup>'
   */
  inline: PropTypes.bool,
  /**
   * Defines if the radio group should display child Radio components inline.
   * @examples '<RadioGroup inlineChildren></RadioGroup>'
   */
  inlineChildren: PropTypes.bool,

  /* Pass inline styling here. */
  style: PropTypes.object
};

export default RadioGroup;
