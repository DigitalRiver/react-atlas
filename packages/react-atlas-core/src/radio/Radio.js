import React, { Component, PropTypes } from 'react';
import { classNames } from '../utils';
import themeable from 'react-themeable';

const Radio = ({className, disabled, inline, name, label, ...props}) => {
  const theme = themeable(props.theme);
  const labelClasses = classNames({
    block: !inline,
    inline,
    disabled
  }, className);

  return (
    <label {...theme(1, ...labelClasses)}>
      <input
        {...props}
        disabled={disabled}
        name={name}
        type="radio"
      />
      {label ? <span> {label}</span> : null}
    </label>
  );
};

Radio.propTypes = {
  defaultChecked: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  inline: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  onFocus: PropTypes.func,
  value: PropTypes.any
};

Radio.defaultProps = {
  className: '',
  disabled: false,
  inline: false
};

Radio.styleguide = {
  category: 'Form Components',
  index: '3.8',
  example: `
<section>
  <h5>Individual Radio Buttons</h5>
  <p>Not aware of each other, use RadioGroup for that.</p>
  <Radio label="Checked Radio" value="checkedRadio" defaultChecked />
  <Radio label="Disabled Radio" value="disabledRadio" disabled/>
  <Radio label="Inlined 1" value="inlined1" className="someClassIMadeUp" inline />
  <Radio label="Inlined 2" value="inlined2" inline />
</section>
`
};

export default Radio;
