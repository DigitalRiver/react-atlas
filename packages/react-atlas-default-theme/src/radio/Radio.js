import { Radio } from "react-atlas-core";
import style from "./radio.css";
import createComponent from "../createComponent";

export default createComponent(Radio, style);
