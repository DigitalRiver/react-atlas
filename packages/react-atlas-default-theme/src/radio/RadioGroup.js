import { RadioGroup } from "react-atlas-core";
import style from "./radio.css";
import createComponent from "../createComponent";

export default createComponent(RadioGroup, style);
