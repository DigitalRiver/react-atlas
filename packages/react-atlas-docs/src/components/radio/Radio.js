import React, { PropTypes } from "react";
import * as Atlas from 'react-atlas';

export default class Radio extends React.Component {
  constructor(props){
    super(props)
  }
  render() {
    return (
      <Atlas.Radio {...this.props}></Atlas.Radio>
    )
  }
}