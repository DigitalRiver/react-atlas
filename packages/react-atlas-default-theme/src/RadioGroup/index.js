import * as RadioGroupStyle from "./RadioGroup.css";
export { RadioGroupStyle };
