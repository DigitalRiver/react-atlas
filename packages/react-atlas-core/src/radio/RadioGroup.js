import React, { PropTypes } from "react";
import Radio from "./Radio";
import themeable from "react-themeable";

const propTypes = {
  "children": PropTypes.node,
  "className": PropTypes.string,
  "name": PropTypes.string,
  "theme": PropTypes.object
};

const defaultProps = {
  "className": ""
};

const RadioGroup = ({ className, children, name, ...props }) => {
  const theme = themeable(props.theme);

  return (
    <div {...props} {...theme(1, className)}>
      {React.Children.map(children, child => {
        if (child.type === Radio) {
          return <Radio {...child.props} name={name} theme={props.theme} />;
        } else {
          return child;
        }
      })}
    </div>
  );
};

RadioGroup.propTypes = propTypes;
RadioGroup.defaultProps = defaultProps;

RadioGroup.styleguide = {
  "category": "Form Components",
  "index": "3.9",
  "example": 
    `
<section>
  <h5>RadioGroup with Radio Buttons</h5>
  <RadioGroup name="comic">
    <Radio label="Checked Radio" value="checkedRadio" defaultChecked />
    <Radio label="Disabled Radio" value="disabledRadio" disabled/>
    <Radio label="Inlined 1" value="inlined1" className="someClassIMadeUp" inline onFocus={this.handleFocus}/>
    <Radio label="Inlined2" value="inlined2" inline onBlur={this.handleBlur}/>
  </RadioGroup>
</section>
`
  
};

export default RadioGroup;
