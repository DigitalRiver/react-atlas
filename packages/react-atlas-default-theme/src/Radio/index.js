import * as RadioStyle from "./radio.css";
export { RadioStyle };
