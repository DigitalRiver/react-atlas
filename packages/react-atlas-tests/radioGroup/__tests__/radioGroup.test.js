import React from "react";
import { mount } from "enzyme";
import { RadioCore } from "../../../react-atlas-core/src/Radio/index";
import { RadioGroupCore } from "../../../react-atlas-core/src/RadioGroup/index";

import renderer from "react-test-renderer";

describe("Test RadioGroup component render", () => {
  it("Render correctly", () => {
    const tree = renderer
      .create(
        <RadioGroupCore inline name="test">
          <RadioCore label="Option 1" value="first" />
          <RadioCore label="Option 2" value="second" />
          <RadioCore label="Option 3" value="third" />
          <RadioCore label="Option 4" value="fourth" />
          <RadioCore label="Option 5" value="fifth" />
          <RadioCore label="Option 6" value="sixth" />
        </RadioGroupCore>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing radio component", () => {
  function _findRadio(node, label) {
    return (
      node.props().type == "radio" &&
      node.props().checked != undefined &&
      node.props().label == label
    );
  }

  it("RadioGroup - Basic test", function() {
    const result = mount(
      <RadioGroupCore inline name="test" title="Title" classname="Class">
        <RadioCore label="Option 1" value="first" />
        <RadioCore label="Option 2" value="second" />
      </RadioGroupCore>
    );
  });

  it("RadioGroup - Change selection test", function() {
    const result = mount(
      <RadioGroupCore inline name="test">
        <RadioCore label="Option 1" value="first" />
        <RadioCore label="Option 2" value="second" />
        <RadioCore label="Option 3" value="third" />
        <RadioCore label="Option 4" value="fourth" />
        <RadioCore label="Option 5" value="fifth" />
        <RadioCore label="Option 6" value="sixth" />
      </RadioGroupCore>
    );
    expect(result.state().checkedRadio).toEqual("first");

    result.findWhere(n => _findRadio(n, "Option 4")).simulate("click");
    expect(result.state().checkedRadio).toEqual("fourth");

    result.findWhere(n => _findRadio(n, "Option 2")).simulate("click");
    expect(result.state().checkedRadio).toEqual("second");
  });

  it("RadioGroup - Change props ", function() {
    const result = mount(
      <RadioGroupCore inline name="test">
        <RadioCore label="Option 1" value="first" />
        <RadioCore label="Option 2" value="second" />
        <RadioCore label="Option 3" value="third" />
        <RadioCore label="Option 4" value="fourth" />
        <RadioCore label="Option 5" value="fifth" />
        <RadioCore label="Option 6" value="sixth" />
      </RadioGroupCore>
    );

    result.setProps({ name: "Name" });

    result.findWhere(n => _findRadio(n, "Option 4")).simulate("click");
    expect(result.state().checkedRadio).toEqual("fourth");

    result.setProps({ selectedValue: "second" });
    expect(result.state().checkedRadio).toEqual("second");
  });
});
