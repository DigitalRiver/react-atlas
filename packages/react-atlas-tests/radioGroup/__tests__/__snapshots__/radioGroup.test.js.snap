// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RadioGroup component render Render correctly 1`] = `
<div
  className=""
  style={undefined}
  styleName="inline"
>
  <div
    onClick={[Function]}
    style={undefined}
    styleName="radio_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <label
        className=""
        styleName="label"
        title="Option 1"
      >
        Option 1
      </label>
      <div
        styleName="float_left"
      >
        <div
          styleName="container"
        >
          <input
            className="ra_Input__max ra_Input__opacity"
            id={undefined}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder={undefined}
            styleName=""
            type="radio"
            value="first"
          />
        </div>
        <div
          styleName="checked"
        >
          <div
            styleName="checkmark"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    onClick={[Function]}
    style={undefined}
    styleName="radio_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <label
        className=""
        styleName="label"
        title="Option 2"
      >
        Option 2
      </label>
      <div
        styleName="float_left"
      >
        <div
          styleName="container"
        >
          <input
            className="ra_Input__max ra_Input__opacity"
            id={undefined}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder={undefined}
            styleName=""
            type="radio"
            value="second"
          />
        </div>
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    onClick={[Function]}
    style={undefined}
    styleName="radio_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <label
        className=""
        styleName="label"
        title="Option 3"
      >
        Option 3
      </label>
      <div
        styleName="float_left"
      >
        <div
          styleName="container"
        >
          <input
            className="ra_Input__max ra_Input__opacity"
            id={undefined}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder={undefined}
            styleName=""
            type="radio"
            value="third"
          />
        </div>
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    onClick={[Function]}
    style={undefined}
    styleName="radio_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <label
        className=""
        styleName="label"
        title="Option 4"
      >
        Option 4
      </label>
      <div
        styleName="float_left"
      >
        <div
          styleName="container"
        >
          <input
            className="ra_Input__max ra_Input__opacity"
            id={undefined}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder={undefined}
            styleName=""
            type="radio"
            value="fourth"
          />
        </div>
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    onClick={[Function]}
    style={undefined}
    styleName="radio_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <label
        className=""
        styleName="label"
        title="Option 5"
      >
        Option 5
      </label>
      <div
        styleName="float_left"
      >
        <div
          styleName="container"
        >
          <input
            className="ra_Input__max ra_Input__opacity"
            id={undefined}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder={undefined}
            styleName=""
            type="radio"
            value="fifth"
          />
        </div>
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    onClick={[Function]}
    style={undefined}
    styleName="radio_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <label
        className=""
        styleName="label"
        title="Option 6"
      >
        Option 6
      </label>
      <div
        styleName="float_left"
      >
        <div
          styleName="container"
        >
          <input
            className="ra_Input__max ra_Input__opacity"
            id={undefined}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder={undefined}
            styleName=""
            type="radio"
            value="sixth"
          />
        </div>
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
</div>
`;
